jerry-core/ecma/builtin-objects/ecma-builtin-object.c

```diff
@@ -746,6 +747,74 @@ ecma_builtin_object_object_get_own_property_descriptor (ecma_object_t *obj_p, /*
   return ECMA_VALUE_UNDEFINED;
 } /* ecma_builtin_object_object_get_own_property_descriptor */
 
+#if ENABLED (JERRY_ESNEXT)
+/**
+ * The Object object's 'getOwnPropertyDescriptors' routine
+ *
+ * See also:
+ *          ECMA-262 v11, 19.1.2.9
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_builtin_object_object_get_own_property_descriptors (ecma_object_t *obj_p) /**< routine's first argument */
+{
+  /* 2 */
+  ecma_collection_t *prop_names_p = ecma_op_object_get_property_names (obj_p, ECMA_LIST_SYMBOLS);
+
+#if ENABLED (JERRY_BUILTIN_PROXY)
+  if (prop_names_p == NULL)
+  {
+    return ECMA_VALUE_ERROR;
+  }
+#endif /* ENABLED (JERRY_BUILTIN_PROXY) */
+
+  ecma_value_t *names_buffer_p = prop_names_p->buffer_p;
+
+  /* 3 */
+  ecma_object_t *object_prototype_p = ecma_builtin_get (ECMA_BUILTIN_ID_OBJECT_PROTOTYPE);
+  ecma_object_t *descriptors_p = ecma_create_object (object_prototype_p, 0, ECMA_OBJECT_TYPE_GENERAL);
+
+  /* 4 */
+  for (uint32_t i = 0; i < prop_names_p->item_count; i++)
+  {
+    ecma_string_t *property_name_p = ecma_get_prop_name_from_value (names_buffer_p[i]);
+
+    /* 4.a */
+    ecma_property_descriptor_t prop_desc;
+    ecma_value_t status = ecma_op_object_get_own_property_descriptor (obj_p, property_name_p, &prop_desc);
+
+    if (ECMA_IS_VALUE_ERROR (status))
```

Comment:
IMO this should throw error only if the proxy is enabled.

Reply:
Good catch! Fixed.